src/main/java/pl/project13/maven/git/build/TeamCityBuildServerData.java

```diff
@@ -0,0 +1,60 @@
+package pl.project13.maven.git.build;
+
+import org.jetbrains.annotations.NotNull;
+import pl.project13.maven.git.GitCommitPropertyConstant;
+import pl.project13.maven.git.log.LoggerBridge;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+public class TeamCityBuildServerData extends BuildServerDataProvider {
+
+  private final Properties systemProperties = new Properties();
+
+  TeamCityBuildServerData(LoggerBridge log) {
+    this(log, System.getenv());
```

Comment:
Mhh I don't like that we here again fetch the system environment variables within the constructor.

Why not initialize the teamcityBuildProperties with null like so:
```private Properties systemProperties = null;```
Whenever it is needed within the functions use ``getTeamcityBuildProperties(env)`` with:
```java
private Properties getTeamcityBuildProperties(@NotNull Map<String, String> env) {
    if (teamcityBuildProperties == null) {
      try {
        teamcityBuildProperties = new Properties();
        teamcityBuildProperties.load(new FileInputStream(env.get("TEAMCITY_BUILD_PROPERTIES_FILE")));
      } catch (IOException e) {
        log.error("Failed to retrieve Teamcity properties file", e);
      }
    }
    return teamcityBuildProperties;
}
```

Could the statement ``new FileInputStream(env.get("TEAMCITY_BUILD_PROPERTIES_FILE"))`` potentially throw a NullPointerException if ``env.get("TEAMCITY_BUILD_PROPERTIES_FILE")`` returns null for some reason??